docs/viplab3.0/computation_result.md

```diff
@@ -1,141 +1,235 @@
 
 Notes:
+
 * There are error messages containing multiple error triggering locations. This leads to multiple info_elements containing same "message" and "output", but different "source" objects. Note: another way would be to specify "sources":[{},...] as alternative to "source":{}.
 * "type": "fatal" may be introduced later
 * Compiler output may contain multiple error positions; only the first one should be extracted.
 
-## System status JSON message
 
-to inform frontend about computations in-preparation...
+##### notification origin actifact
+|Key            |Value Type |Opt / Must |Description |Comment | AS |
+|---------------|-----------|-----------|------------|--------|----|
+|source | string | must | identifier of element from Solution/Exercise elements with source *input*, which has triggered (error/warning/...) message |
+|extract | string | must | Part of referenced element. Extracted from message triggering input. |
+|begin | unsigned int | must | Offset of extract into ... |
+|end | unsigned int | must | ... referenced element: C-like (zero-based). |
+|line |1..uint_max | must | Location of error/warning ... |
+|col |1..uint_max | opt | ... referring to position in triggering input element. | It is optional, since there are messages containing line info only |
+
+##### notification output actifact
+|Key            |Value Type |Opt / Must |Description |Comment | AS |
+|---------------|-----------|-----------|------------|--------|----|
+|source | string | must | identifier of element from Result elements with interpreter/compiler *output*, which contains error/warning message |
+|extract | string | must | Part of referenced element. Extracted from message triggering input. |
+|begin | unsigned int | must | Offset of extract into ... |
+|end | unsigned int | must | ... referenced element: C-like (zero-based). |
+
+
+## Leftovers from Viplab 2.0
+
+might be converted later, just keep around as inspration
+
+|Key            |Value Type |Opt / Must |Description |Comment |
+|---------------|-----------|-----------|------------|--------|
+|startTime  | datetimeString | must | May be local or universal time. | Just after getting a Solution.|
+|duration   | durationString | must | e.g. "128ms" | Computation duration time since startTime (increases with each intermediate Result). |
+|finishTime | datetimeString | must, iff status == "final" | May be local or universal time. | Just before posting final Result.|
+|CC_versionLong | string | must | | |
+|CC_version  | string | must | | |
+|chain_version  | string | must | | |
+|technicalInfo | struct | must | | |
+|technicalInfo/host | string | must | CC host | |
+|technicalInfo/PID | unsigned | must | CC PID | |
+|technicalInfo/&lt;key&gt; | &lt;val_type&gt; | opt | &lt;key&gt;: locally unique, &lt;val_type&gt;: arbitrary type | For allowing further key/val pairs if there should be a need.|
```

Comment:
can be put into metadata

Reply:
I would rather define a debuginfo artifact, in which we could add this information. This way, we have a defined place at which we can put such information.